### DISCUSSION TITLE
Large file upload error in production - "Your request was too large to submit successfully"

### DISCUSSION THREAD

**Discussion by ab-h-i-n** [payloadcms/payload#9205] [Q&A] [answered]
I'm encountering an issue when trying to upload a large (70 MB) video file to a media collection in production with Payload CMS. The error message says, "Your request was too large to submit successfully." However, this issue doesn’t occur in my local environment, where uploads work as expected.

**Payload CMS version: 3.0.0-beta.127
Storage adapter: S3 adapter
Upload limit configured in Payload:**

`
upload: {
    limits: {
        fileSize: 2 * 1024 * 1024 * 1024, // 2GB limit
    }
}

`
**What I've tried so far:**
- Configured Payload's file upload limit to 2GB.

**Comment by vladvlasov256** [comment]
How do you deploy your CMS on production? Do you use Nginx or another proxy?

**Reply by ab-h-i-n** [reply]
I used vercel for hosting

**Reply by vladvlasov256** [reply, marked as answer]
Please, check the request body size limit in Vercel. Are you on the free plan, by the way?

We upload ~200Mb files to Payload but we use Nginx and we have to configure it.

**Reply by ab-h-i-n** [reply]
Thank you for pointing that out! Yes, I am using Vercel for hosting, and I am currently on the free plan. I believe the request body size limit on Vercel’s free plan is 4 MB by default.I might need to look into alternative approaches.
